### ISSUE TITLE
Cannot emit from disconnect handler

### ISSUE THREAD

**Issue by miguelgrinberg** [miguelgrinberg/Flask-SocketIO#201]
See https://github.com/miguelgrinberg/python-socketio/issues/11 for details.

**Comment by peacehh** [comment]
There is another problem when emitting from disconnect handlers: The emit() method leads to a disconnect event trigger in engineio.socket.Socket#send, when a client is gone (that is when `time.time() - self.last_ping > self.server.ping_interval * 5 / 4`). This can cause a recursion that ends in a RuntimeError "maximum recursion depth exceeded".

My workaround is to work in a specific namespace with an empty dummy disconnect handler.The actual disconnect handler is bound to the root namespace. When disconnecting, socketio calls both handlers, but the emit() call triggers the dummy event, so that the recursion is avoided.
```
@socketio.on('connect', namespace='/clients')
def register_connection():
    # do register stuff
    socketio.emit('registered clients', data, namespace='/clients', broadcast=True)

@socketio.on('disconnect')
def unregister_connection():
    # do unregister stuff
    socketio.emit('registered clients', data, namespace='/clients', broadcast=True)

@socketio.on('disconnect', namespace='/clients')
def unregister_connection_dummy():
    pass
```

**Comment by TheDialord** [comment]
Finally i've found my problem . . .

**Comment by miguelgrinberg** [comment]
Fixed in python-socketio package, version 1.1. Please let me know if you continue to see problems after upgrade to that release.

**Comment by kaihil** [comment]
@miguelgrinberg: At a first glance it seems to work now without the workaround, great! I'll let you know if my first impression was wrong ;)  Thanks for your great work, Miguel!
